cmake/catkin_package.cmake

```diff
@@ -374,6 +377,11 @@ function(_catkin_package)
       endif()
     endforeach()
 
+    set(PKG_COMPILE_OPTIONS ${${PROJECT_NAME}_COMPILE_OPTIONS})
+    separate_arguments(PKG_COMPILE_OPTIONS)
+    separate_arguments(PROJECT_COMPILE_OPTIONS)
```

Comment:
Since this is a user provided variable I don't think it is a good idea to mutate it. This should store the modified value in a local variable instead.